test/test_basic_submission.py

```diff
@@ -47,11 +46,11 @@ def setUpClass(cls):
         gcloud_cred_dir = os.path.expanduser('~/.config/gcloud')
         if not os.path.exists(gcloud_cred_dir):
             os.makedirs(gcloud_cred_dir, exist_ok=True)
-        try:
-            shutil.copy(os.environ['TEST_MULE_CREDENTIALS'],
-                        os.path.expanduser('~/.config/gcloud/application_default_credentials.json'))
-        except shutil.SameFileError:
-            pass
+        with open(os.path.expanduser('~/.config/gcloud/application_default_credentials.json'), 'w') as f:
+            f.write(json.dumps({'client_id': os.environ['TEST_MULE_CLIENT_ID'],
+                                'client_secret': os.environ['TEST_MULE_CLIENT'],
+                                'refresh_token': os.environ['TEST_MULE_REFRESH_TOKEN'],
+                                'type': os.environ['TEST_MULE_TYPE']}))
```

Comment:
Instead of reconstructing the credentials file from a bunch of env vars, consider base64 encoding the credentials file using a single env var.

in bash:
```
cat my-creds.json | base64
```

Reply:
Good advice, thanks!